### PR TITLE
ME-2780 auto fallback to websocket transport

### PR DESCRIPTION
**Pull request by atoonk**
# Description

we’ve seen customer enviroments where clients cannot talk to our servers without the need for a websocket proxy. Typically due to strict firewalls
this is to make that easier, less flags, and auto fallback to websocket in case of reachability issues.

the  `-w`, or `--wsproxy` flag is now a bool, and no longer needs a full websocket url.

This is the behaviour where the initial connector doesn't work and we fall back to websocket. Making it easier for users to "discover" this capability.

```
$ border0 client ssh -l root ssh-andree-docker-test-1712177406-myorgblablabla.border0.io
============ Border0 Connection Notice ============
Direct connection failed. This is possibly due to firewall restrictions in your network.
Error: dial tcp 15.197.134.180:20252: i/o timeout
Falling back to using the Border0 WebSocket proxy.
To default to the WebSocket proxy and avoid this message, use --wsproxy or set the BORDER0_WS_PROXY environment variable:
export BORDER0_WS_PROXY=wss://ws.border0.com/ws
==================================================


Connecting to Server: ssh-andree-docker-test-1712177406-border0-demo.border0.io:20252 as root
Welcome to Ubuntu 23.10 (GNU/Linux 6.5.0-26-generic x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/pro

  System information as of Thu Apr  4 20:25:22 UTC 2024

  System load:  0.0               Processes:             98
  Usage of /:   8.0% of 66.76GB   Users logged in:       0
  Memory usage: 21%               IPv4 address for eth0: 146.190.58.74
  Swap usage:   0%                IPv4 address for eth0: 10.48.0.5

0 updates can be applied immediately.


Last login: Thu Apr  4 20:25:48 UTC 2024 from Border0 on pts/0
root@andree-docker-test:~#
root@andree-docker-test:~#
```

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: internal/client/client.go [hunk 1]
```diff
@@ -903,21 +905,56 @@ func StartConnectorAuthListener(hostname string, port int, certificate tls.Certi
...
+			// and suggest setting the env var to avoid this message
+			helpString := ""
+			if runtime.GOOS == "windows" {
+				helpString = "set BORDER0_WS_PROXY=" + defaultWSProxy
+			} else {
+				helpString = "export BORDER0_WS_PROXY=" + defaultWSProxy
+			}
+			fmt.Printf("============ Border0 Connection Notice ============\n"+
+				"Direct connection failed. This is possibly due to firewall restrictions in your network.\n"+
+				"Error: "+err.Error()+"\n"+
```
Comment (btoonk) [inline internal/client/client.go:941]: replace  `+err.Error()+` with %s, and put the err.Error() as an parameter just like helpString.

**Review by btoonk — APPROVED** [review]
Left one comment, lgtm!

File: internal/client/client.go [hunk 2]
```diff
@@ -903,21 +905,56 @@ func StartConnectorAuthListener(hostname string, port int, certificate tls.Certi
...
 		if err != nil {
-			return nil, err
+			// retry with ws proxy, we'll retry on any error
+			// This is to handle the case where the proxy is not reachable
+			// Yay for firewalls and zscalers
+
+			// Inform the user that we are falling back to the ws proxy
+			// and suggest setting the env var to avoid this message
+			helpString := ""
+			if runtime.GOOS == "windows" {
```
Comment (adrianosela) [inline internal/client/client.go:934]: Use format strings instead of string addition.

Also prefer multiple fmt.Println over printf with newlines

**Review by adrianosela — APPROVED** [review]
Nit comments left. Lgtm!

File: internal/client/client.go (hunk 1 above)
Comment (atoonk) [reply internal/client/client.go]: done

File: internal/client/client.go (hunk 2 above)
Comment (atoonk) [reply internal/client/client.go:934]: done, now using format strings instead of string addition.

**Pull request merged**